native/SampleApps/SmartSyncExplorer/RecentContactsTodayExtension/TodayViewController.m

```diff
@@ -0,0 +1,151 @@
+
+@interface TodayViewController () <NCWidgetProviding,SFRestDelegate,UITableViewDelegate,UITableViewDataSource>{
+      NSMutableArray *_contacts;
+}
+@property (weak, nonatomic) IBOutlet UITableView *todayTableView;
+@property (nonatomic, strong) SObjectDataManager *dataMgr;
+
+@end
+
+@implementation TodayViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    #if defined(DEBUG)
+        [SFLogger sharedLogger].logLevel = SFLogLevelDebug;
+    #else
+        [SFLogger sharedLogger].logLevel = SFLogLevelInfo;
+    #endif
+
+    _contacts = [NSMutableArray new];
+    [self.todayTableView setDataSource:self];
+    [self.todayTableView setDelegate:self];
+}
+
+- (void) refreshList {
+    [self.todayTableView reloadData];
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+- (void)widgetPerformUpdateWithCompletionHandler:(void (^)(NCUpdateResult))completionHandler {
+    [SFSDKDatasharingHelper sharedInstance].appGroupName = @"group.com.salesforce.mobilesdk.internal.SmartSyncExplorer";
+    [SFSDKDatasharingHelper sharedInstance].appGroupEnabled = YES;
+    if( [self userLoginStatus] ) {
+       
+        [SalesforceSDKManager setInstanceClass:[SalesforceSDKManagerWithSmartStore class]];
+        [SalesforceSDKManager sharedManager].connectedAppId = RemoteAccessConsumerKey;
+        [SalesforceSDKManager sharedManager].connectedAppCallbackUri = OAuthRedirectURI;
+        [SalesforceSDKManager sharedManager].authScopes = @[@"web", @"api"];
+        [SalesforceSDKManager sharedManager].authenticateAtLaunch = FALSE;
+        
+        NSArray *accounts = [[SFUserAccountManager sharedInstance] allUserAccounts];
+        if(accounts && accounts.count>0) {
+            // check for actual Account or id of account or pick the first one?
+            [[SFUserAccountManager sharedInstance] setCurrentUser:accounts[0]];
+            
+            __weak typeof(self) weakSelf = self;
+            void (^completionBlock)(void) = ^{
+                [weakSelf refreshList];
+            };
+          
+            if (!self.dataMgr) {
+                self.dataMgr = [[SObjectDataManager alloc] initWithDataSpec:[ContactSObjectData dataSpec]];
+            }
+            [self.dataMgr lastModifiedRecords:3 completion:completionBlock];
+            
+        }
+    }
+    completionHandler(NCUpdateResultNewData);
+}
+
+- (BOOL)userLoginStatus {
+    return [[NSUserDefaults msdkUserDefaults] boolForKey:@"userLoggedIn"];
+}
+
+- (void)request:(SFRestRequest *)request didLoadResponse:(id)dataResponse {
+    NSLog(@"Request Success");
+    dispatch_async(dispatch_get_main_queue(), ^{
+        NSArray *records = ((NSDictionary *)dataResponse)[@"records"];
+        [_contacts removeAllObjects];
+        [_contacts addObjectsFromArray:records];
+        [_todayTableView reloadData];
+        NSLog(@"didLoadResponse %@",records);
```

Comment:
Use `SFLogger` instead of `NSLog`. In this case, I don't think we should log all records.